util/stream.go

```diff
@@ -64,9 +67,8 @@ func NewMessageStream(conn net.Conn, parser Parser) *MessageStream {
 	go m.outbound()
 	go m.inbound()
 
-	for i := 0; i < 25; i++ {
```

Comment:
This code seems to have a lot to do with the channel based buffer pool. It wasn't a good idea in the end. The messages would lose their order.